backend/src/graphql/resolver/ContributionResolver.ts

```diff
@@ -58,19 +74,31 @@ export class ContributionResolver {
     @Arg('id', () => Int) id: number,
     @Ctx() context: Context,
   ): Promise<boolean> {
+    const event = new Event()
     const user = getUser(context)
     const contribution = await dbContribution.findOne(id)
     if (!contribution) {
+      logger.error('Contribution not found for given id')
       throw new Error('Contribution not found for given id.')
     }
     if (contribution.userId !== user.id) {
+      logger.error('Can not delete contribution of another user')
       throw new Error('Can not delete contribution of another user')
     }
     if (contribution.confirmedAt) {
+      logger.error('A confirmed contribution can not be deleted')
       throw new Error('A confirmed contribution can not be deleted')
     }
+
     contribution.contributionStatus = ContributionStatus.DELETED
+    contribution.deletedAt = new Date()
     await contribution.save()
+
+    const eventDeleteContribution = new EventContributionDelete()
+    eventDeleteContribution.userId = user.id
+    eventDeleteContribution.contributionId = contribution.id
+    await eventProtocol.writeEvent(event.setEventContributionDelete(eventDeleteContribution))
```

Comment:
in case the attribute amount is mandatory, as requested, then here the initialization is missing

Reply:
solved, true